pkg/objstore/objstore.go

```diff
@@ -437,6 +437,87 @@ func (b *metricBucket) Name() string {
 	return b.bkt.Name()
 }
 
+type prefixedBucket struct {
+	bkt    Bucket
```

Comment:
How about embedding `Bucket` into this struct so that you don't have to implement all the passthrough functions?

Reply:
Yes, this would work well :+1: The downside is that we can miss prefix for newly added method but we should not need to optimize for this

Reply:
resolving this as their are no action points to address